apps/Hangover/install-64

```diff
@@ -0,0 +1,305 @@
+if [[ "$PAGE_SIZE" == "16384" ]]; then
+  #switch to 4K pagesize kernel
+  if [ -f /boot/config.txt ] || [ -f /boot/firmware/config.txt ]; then
+    if [ -f /boot/firmware/config.txt ]; then
+      boot_config="/boot/firmware/config.txt"
+    elif [ -f /boot/config.txt ]; then
+      boot_config="/boot/config.txt"
+    fi
+    text="Raspberry Pi 5 PiOS images ship by default with a 16K PageSize Linux Kernel.
+This kernel causes incompatibilities with some software including Wine https://github.com/raspberrypi/bookworm-feedback/issues/107
+
+Would you like to automatically switch to a 4K PageSize Linux Kernel?"
+    userinput_func "$text" "No, keep 16K PageSize Kernel and Exit" "Yes, switch to 4K PageSize Kernel"
+    if [ "$output" == "No, keep 16K PageSize Kernel and Exit" ]; then
+      error "User error: Your current running kernel is built with 16K PageSize and is incompatible with Wine (x64) with Box64. You must switch to a 4K PageSize kernel (and chose to not do so automatically) before installing Wine (x64)."
+    fi
+    echo "" | sudo tee --append $boot_config >/dev/null
+    echo "[pi5]" | sudo tee --append $boot_config >/dev/null
+    echo "kernel=kernel8.img" | sudo tee --append $boot_config >/dev/null
+    echo -e "The 4K PageSize Kernel has been enabled by adding 'kernel=kernel8.img' to $boot_config\nPlease reboot now and install the Wine (x64) app again."
+    sleep infinity
+  else
+    error "User error (reporting allowed): Your current running kernel is built with 16K PageSize and is incompatible with Wine (x64) with Box64. Changing kernels automatically cannot be done since no /boot/config.txt or /boot/firmware/config.txt file was found."
+  fi
+fi
+
+if [ "$__os_codename" == "bullseye" ]; then
+  ho_distro="debian11"
+elif [ "$__os_codename" == "bookworm" ]; then
+  ho_distro="debian12"
+elif [ "$__os_codename" == "focal" ]; then
+  ho_distro="ubuntu2004"
+elif [ "$__os_codename" == "jammy" ]; then
+  ho_distro="ubuntu2204"
+elif [ "$__os_codename" == "noble" ]; then
+  ho_distro="ubuntu2404"
+else
+  error "User error: You are not using a supported Pi-Apps distribution."
+fi
+
+cd /tmp || error "Could not move to /tmp folder"
+wget https://github.com/AndreRH/hangover/releases/download/hangover-${version}/hangover_${version}_${ho_distro}_${__os_codename}_arm64.tar || error "Failed to download Hangover!"
+tar -xf hangover_${version}_${ho_distro}_${__os_codename}_arm64.tar || error "Failed to extract Hangover!"
+rm -f hangover_${version}_${ho_distro}_${__os_codename}_arm64.tar
+install_packages /tmp/hangover-libarm64ecfex_${version}_arm64.deb /tmp/hangover-libqemu_${version}~${__os_codename}_arm64.deb /tmp/hangover-libwow64fex_${version}_arm64.deb /tmp/hangover-wine_${version}~${__os_codename}_arm64.deb || exit 1
+rm -f ./hangover-libarm64ecfex_${version}_arm64.deb ./hangover-libqemu_${version}~${__os_codename}_arm64.deb ./hangover-libwow64fex_${version}_arm64.deb ./hangover-wine_${version}~${__os_codename}_arm64.deb
```

Comment:
this will conflict with the existing `Wine (x86)` and `Wine (x64)`. At pi-apps we don't allow applications to break each other. If they must break then installing one must automatically uninstall another one (eg: https://github.com/Botspot/pi-apps/blob/1c2387ba5c528ae3c78e35d87aa93d5e68ed5f4a/apps/AndroidBuddy/uninstall#L4 )

Please consider using a different naming convention for `hangover` than `wine` to avoid the conflict.

Reply:
I'd expect the uninstall to happen in the install of Hangover then, no?

Reply:
Yes, if currently installed it will uninstall during the Hangover installation.

Reply:
pushed, though IMHO it can't conflict with `Wine (x86)`, cause it's 32-bit only and Hangover is 64-bit only

Reply:
The binary name for using `Wine (x86)` does conflict as both `Wine (x86)` and `Wine (x64)` use the binary name `wine` (as per upstream defaults)

Reply:
AFAICT `Wine (x86)` can't be installed on 64-bit systems, Hangover can't be installed on a 32-bit system, so both will never meet, no?

Reply:
We also have Wine (x64) which also has a wine command, which is installable on arm64 systems. It would seem to conflict.

Reply:
Now that statement is true. So its not needed for `Wine (x86)` because of that.

Reply:
But that's not a bad thing. It would likely be a bad idea for a user to have two wine versions installed and running at the same time anyway.